src/Npgsql/NpgsqlConnection.cs

```diff
@@ -993,7 +993,25 @@ public int ProcessID
         /// <remarks>
         /// See https://www.postgresql.org/docs/current/static/sql-copy.html.
         /// </remarks>
-        public NpgsqlBinaryImporter BeginBinaryImport(string copyFromCommand)
+        public NpgsqlBinaryImporter BeginBinaryImport(string copyFromCommand) =>
+            BeginBinaryImport(copyFromCommand, async: false, CancellationToken.None).GetAwaiter().GetResult();
+
+        /// <summary>
+        /// Begins a binary COPY FROM STDIN operation, a high-performance data import mechanism to a PostgreSQL table.
+        /// </summary>
+        /// <param name="copyFromCommand">A COPY FROM STDIN SQL command</param>
+        /// <param name="cancellationToken">An optional token to cancel the asynchronous operation. The default value is None.</param>
+        /// <returns>A <see cref="NpgsqlBinaryImporter"/> which can be used to write rows and columns</returns>
+        /// <remarks>
+        /// See https://www.postgresql.org/docs/current/static/sql-copy.html.
+        /// </remarks>
+        public Task<NpgsqlBinaryImporter> BeginBinaryImportAsync(string copyFromCommand, CancellationToken cancellationToken = default)
+        {
+            using (NoSynchronizationContextScope.Enter())
+                return BeginBinaryImport(copyFromCommand, async: true, cancellationToken);
+        }
+
+        async Task<NpgsqlBinaryImporter> BeginBinaryImport(string copyFromCommand, bool async, CancellationToken cancellationToken = default)
```

Comment:
And `ValueTask`s are cheaper in case of state machines and starting .NET 5 may cost nothing due to [pooling](https://devblogs.microsoft.com/dotnet/async-valuetask-pooling-in-net-5/). It's opt-in for now and the feature under testing in production, so it may be removed in the future (see https://github.com/dotnet/runtime/issues/13633). But value tasks still are cheaper that tasks returning result.